documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,954 @@
+
+// Once fully finished with the Mqtt5Client:
+client.close();
+```
+
+```java
+// client.close() will be called at the end of the try/catch block.
+try (Mqtt5CLient client = builder.build()) {
+    // ...
+} catch (Exception e) {
+}
+
+```
+
+### Reconnects
+
+The v1 SDK has a maximum number of retry attempts for auto-reconnect. If you exceed the maximum number of retries, the v1 SDK will throw
+a permanent error and you will not be able to use the same client instance again.
+
+The v2 SDK attempts to reconnect automatically until connection succeeds or `client.stop()` is called. The reconnection parameters,
+such as min/max delays and [jitter modes](https://awslabs.github.io/aws-crt-java/software/amazon/awssdk/crt/io/ExponentialBackoffRetryOptions.JitterMode.html),
+are configurable through [`AwsIotMqtt5ClientBuilder`](https://aws.github.io/aws-iot-device-sdk-java-v2/software/amazon/awssdk/iot/AwsIotMqtt5ClientBuilder.html).
+
+#### Example of tweaking reconnection settings in the v1 SDK
+
+```java
+client.setBaseRetryDelay(1000L);
+client.setMaxRetryDelay(10000L);
+```
+
+#### Example of tweaking reconnection settings in the v2 SDK
+
+```java
+clientBuilder.withMinReconnectDelayMs(1000L);
+clientBuilder.withMaxReconnectDelayMs(10000L);
+clientBuilder.withRetryJitterMode(JitterMode.Full);
+clientBuilder.withMinConnectedTimeToResetReconnectDelayMs(5000L);
+Mqtt5Client client = clientBuidler.build();
+```
+
+### Offline operations queue
+
+In the v1 SDK, if you have too many in-flight QoS 1 messages, you will encounter the `too many publishes in Progress` error
+on publishing messages. This is caused by the so-called [in-flight publish limit](https://github.com/aws/aws-iot-device-sdk-java/blob/master/README.md#increase-in-flight-publish-limit-too-many-publishes-in-progress-error).
+By default, the v1 SDK supports a maximum of 10 in-flight operations.
+
+The v2 SDK does not limit the number of in-flight messages. Additionally, v2 provides a way to configure which kind of packets
+will be placed into the offline queue when the client is in the disconnected state. The following code snippet demonstrates
+how to enable storing all packets except QOS0 publish packets in the offline queue on disconnect:
+
+#### Example of configuring the offline queue in the v2 SDK
+
+```java
+AwsIotMqtt5ClientBuilder builder = AwsIotMqtt5ClientBuilder.newDirectMqttBuilderWithMtlsFromPath(/*...*/);
+builder.withOfflineQueueBehavior(ClientOfflineQueueBehavior.FAIL_QOS0_PUBLISH_ON_DISCONNECT);
+Mqtt5Client client = builder.build();
+```
+
+> [!NOTE]  
+> AWS IoT Core [limits the number of allowed operations per second](https://docs.aws.amazon.com/general/latest/gr/iot-core.html#message-broker-limits).
+The [`getOperationStatistics`](https://awslabs.github.io/aws-crt-java/software/amazon/awssdk/crt/mqtt5/Mqtt5Client.html#getOperationStatistics())
+method returns the current state of an `Mqtt5Client` object's queue of operations, which may help with tracking the number
```

Comment:
those single quotes are messing with the formatting/highlighting i think.. we might need to escape it